packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -859,12 +862,397 @@ class ResizeImageKey {
     }
     return other is ResizeImageKey
         && other._providerCacheKey == _providerCacheKey
+        && other._policy == _policy
         && other._width == _width
-        && other._height == _height;
+        && other._height == _height
+        && other._allowUpscaling == _allowUpscaling;
   }
 
   @override
-  int get hashCode => Object.hash(_providerCacheKey, _width, _height);
+  int get hashCode => Object.hash(_providerCacheKey, _policy, _width, _height, _allowUpscaling);
+}
+
+/// Configures the behavior for [ResizeImage].
+///
+/// This is used in [ResizeImage.policy] to affect how the [ResizeImage.width]
+/// and [ResizeImage.height] properties are interpreted.
+enum ResizeImagePolicy {
+  /// Sizes the image to the exact width and height specified by
+  /// [ResizeImage.width] and [ResizeImage.height].
+  ///
+  /// If [ResizeImage.width] and [ResizeImage.height] are both non-null, the
+  /// output image will have the specified width and height, stretching the
```

Comment:
It'd be nice to be more specific about how the new aspect ratio is applied.

Maybe

> output image will have the spedified width and height, with the aspect ratio of the new width and height regardless of whether it matches the intrinsic aspect ratio of the image's width and height.